File: android/src/main/java/com/reactnativecommunity/viewpager/BiMap.java
package com.reactnativecommunity.viewpager;

import java.util.HashMap;
import java.util.Map;

public class BiMap<K, V> {
    private final Map<K, V> keyToValue = new HashMap<>();
    private final Map<V, K> valueToKey = new HashMap<>();

    public void clear() {
        keyToValue.clear();
        valueToKey.clear();
    }

    public V getByKey(K key) {
        return keyToValue.get(key);
    }

    public K getByValue(V value) {
        return valueToKey.get(value);
    }

    public void put(K key, V value) {
        if (keyToValue.put(key, value) != null || valueToKey.put(value, key) != null) {
            // Note: BiMap is now in an undefined state.
            throw new IllegalStateException("Existing BiMap entry overwritten.");
        }
    }
}

File: android/src/main/java/com/reactnativepagerview/BiMap.kt
package com.reactnativepagerview

class BiMap<K, V> {
    private val keyToValue = mutableMapOf<K, V>()
    private val valueToKey = mutableMapOf<V, K>()

    fun clear() {
        keyToValue.clear()
        valueToKey.clear()
    }

    fun getByKey(key: K): V? {
        return keyToValue[key]
    }

    fun getByValue(value: V): K? {
        return valueToKey[value]
    }

    fun put(key: K, value: V) {
        check(keyToValue.put(key, value) == null && valueToKey.put(value, key) == null) {
            // Note: BiMap is now in an undefined state.
            "Existing BiMap entry overwritten."
        }
    }
}
